celery/app/task.py

```diff
@@ -972,6 +972,18 @@ def update_state(self, task_id=None, state=None, meta=None, **kwargs):
         self.backend.store_result(
             task_id, meta, state, request=self.request, **kwargs)
 
+    def before_start(self, task_id, args, kwargs):
+        """Handler called before the task starts.
+
+        Arguments:
+            task_id (str): Unique id of the task to execute.
+            args (Tuple): Original arguments for the retried task.
+            kwargs (Dict): Original keyword arguments for the retried task.
```

Comment:
The documentation is wrong here.
This is called for every task, not just retried tasks.

Reply:
Updated.